test/minikube/minikube_rolling_upgrade_test.go

```diff
@@ -67,28 +71,74 @@ func TestAdminReadinessProbe(t *testing.T) {
 	output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", admin1, "-c", "admin", "--", "readinessprobe")
 	require.NoError(t, err, "readinessprobe failed: %s", output)
 
-	// delete Raft log on admin-0 and kill admin-0 so that it bootstraps a
+	// make sure direct invocation of livenessprobe script succeeds on both
+	// Admin processes
+	output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", admin0, "-c", "admin", "--", "livenessprobe")
+	require.NoError(t, err, "livenessprobe failed: %s", output)
+	output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", admin1, "-c", "admin", "--", "livenessprobe")
+	require.NoError(t, err, "livenessprobe failed: %s", output)
+
+	// get restart count for admin-1
+	pod, err := testlib.FindPod(t, namespaceName, admin1)
+	require.NoError(t, err)
+	var restartCountBefore int32 = -1
+	for _, containerStatus := range pod.Status.ContainerStatuses {
+		if containerStatus.Name == "admin" {
+			restartCountBefore = containerStatus.RestartCount
+		}
+	}
+	require.NotEqual(t, -1, restartCountBefore, "Unable to get restart count for admin container")
+
+	// delete raftlog PVC and pod for admin-0 so that it bootstraps a
 	// disjoint domain when it is restarted and refuses messages from
 	// admin-1
-	k8s.RunKubectl(t, options, "exec", admin0, "-c", "admin", "--", "rm", "-f", "/var/opt/nuodb/raftlog")
-	k8s.RunKubectl(t, options, "delete", "pod", admin0)
+	k8s.RunKubectl(t, options, "delete", "pvc/raftlog-"+admin0, "pod/"+admin0)
 
 	// make sure readinessprobe on admin-1 eventually fails, either because
 	// there is no leader for it to converge with or because it thinks it is
 	// the leader but is not connected to a quorum
 	testlib.Await(t, func() bool {
-		// make sure readinessprobe fails
-		output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", admin1, "-c", "admin", "--", "readinessprobe")
-		code, err := shell.GetExitCodeForRunCommandError(err)
-		require.NoError(t, err)
-		if code == 0 {
-			return false
+		// make sure readinessprobe and livenessprobe fail by invoking
+		// them directly
+		for _, probe := range []string{"readinessprobe", "livenessprobe"} {
+			output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", admin1, "-c", "admin", "--", probe)
```

Comment:
Is there a timing issue here because the liveness probe might have restarted the container already due to `admin.livenessProbe.failureThreshold=1`?

Reply:
Good point. I will just remove these checks since they are redundant with the status checks that are performed below.